pkg/config/pipeline_builder_test.go

```diff
@@ -0,0 +1,147 @@
+
+	b, err = json.Marshal(params[2])
+	require.NoError(t, err)
+	require.Equal(t, `{"name":"loki","write":{"type":"loki","loki":{"url":"http://loki:3100/","batchWait":"1s","batchSize":102400,"timeout":"10s","minBackoff":"1s","maxBackoff":"5m","maxRetries":10,"timestampLabel":"TimeReceived","timestampScale":"1s"}}}`, string(b))
+}
+
+func TestKafkaPromPipeline(t *testing.T) {
+	pl := NewKafkaPipeline("ingest", api.IngestKafka{
+		Brokers: []string{"http://kafka"},
+		Topic:   "netflows",
+		GroupId: "my-group",
+	})
+	pl = pl.TransformFilter("filter", api.TransformFilter{
+		Rules: []api.TransformFilterRule{{
+			Type:  "remove_entry_if_doesnt_exist",
+			Input: "doesnt_exist",
+		}},
+	})
+	pl = pl.Aggregate("aggregate", []api.AggregateDefinition{{
+		Name:      "src_as_connection_count",
+		By:        api.AggregateBy{"srcAS"},
+		Operation: "count",
+	}})
+	pl = pl.EncodePrometheus("prom", api.PromEncode{
+		Metrics: api.PromMetricsItems{{
+			Name: "connections_per_source_as",
+			Type: "counter",
+			Filter: api.PromMetricsFilter{
+				Key:   "name",
+				Value: "src_as_connection_count",
+			},
+			ValueKey: "recent_count",
+			Labels:   []string{"by", "aggregate"},
+			Buckets:  []float64{},
+		}},
+		Port:   9090,
+		Prefix: "flp_",
+	})
+	stages := pl.GetStages()
+	require.Len(t, stages, 4)
+
+	b, err := json.Marshal(stages)
+	require.NoError(t, err)
+	require.Equal(t, `[{"name":"ingest"},{"name":"filter","follows":"ingest"},{"name":"aggregate","follows":"filter"},{"name":"prom","follows":"aggregate"}]`, string(b))
+
+	params := pl.GetStageParams()
+	require.Len(t, params, 4)
+
+	b, err = json.Marshal(params[0])
+	require.NoError(t, err)
+	require.Equal(t, `{"name":"ingest","ingest":{"type":"kafka","kafka":{"brokers":["http://kafka"],"topic":"netflows","groupid":"my-group","groupBalancers":null,"startOffset":"","batchReadTimeout":0}}}`, string(b))
+
+	b, err = json.Marshal(params[1])
+	require.NoError(t, err)
+	require.Equal(t, `{"name":"filter","transform":{"type":"filter","filter":{"rules":[{"input":"doesnt_exist","type":"remove_entry_if_doesnt_exist"}]}}}`, string(b))
+
+	b, err = json.Marshal(params[2])
+	require.NoError(t, err)
+	require.Equal(t, `{"name":"aggregate","extract":{"type":"aggregates","aggregates":[{"name":"src_as_connection_count","by":["srcAS"],"operation":"count","recordKey":"","topK":0}]}}`, string(b))
+
+	b, err = json.Marshal(params[3])
+	require.NoError(t, err)
+	require.Equal(t, `{"name":"prom","encode":{"type":"prom","prom":{"metrics":[{"name":"connections_per_source_as","type":"counter","filter":{"key":"name","value":"src_as_connection_count"},"valueKey":"recent_count","labels":["by","aggregate"],"buckets":[]}],"port":9090,"prefix":"flp_","expiryTime":0}}}`, string(b))
+}
+
+func TestForkPipeline(t *testing.T) {
```

Comment:
it's because both `WriteLoki` and `WriteStdout` are called from the same `plFork` instance, so they take the same `follows` stage, resulting in a fork. If I wanted to chain `WriteLoki` and `WriteStdout` (ok, even if it doesn't make sense and would produce an error at runtime) I would have written:

```go
  next := plFork.WriteLoki("loki", api.GetWriteLokiDefaults())
  next.WriteStdout("stdout", api.WriteStdout{})
```

I guess I should add some doc